### PR TITLE
WIP  DOC: add high-level overview of multi-level decompositions

### PR DESCRIPTION
**Pull request by grlee77**
This adds a page to the documentation that illustrates the difference between `wavedecn`, `fswt` and wavelet packets.  Still needs some links/etc and should not be merged until after #204.

I created the images in [Inkscape](https://inkscape.org).

### CODE REVIEW

**Comment by rgommers** [comment]
Nice, looks like a good addition to the docs.

File: doc/source/ref/2d-decompositions-overview.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+manner.
+
+
+Standard Multilevel Discrete Wavelet Transform
+----------------------------------------------
+
+The most common approach to the multilevel discrete wavelet transform involves
+further decomposition of only the approximation subband at each subsequent
+level.  This is also sometimes referred to as the Mallat decomposition
+[TODO: REF].  In 2D the discrete wavelet transform, produces four sets of
```
Comment (rgommers) [inline doc/source/ref/2d-decompositions-overview.rst]: https://ieeexplore.ieee.org/document/192463/ I think

File: doc/source/ref/2d-decompositions-overview.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+coefficients corresponding to the 4 possible compinations of the wavelet
+decomposition filters over the two separate axes.  (In n-dimensions, there
+are 2**n sets of coefficients).  This approach is illustrated below.
```
Comment (rgommers) [inline doc/source/ref/2d-decompositions-overview.rst]: formatting: double backticks around `2**n`

File: doc/source/ref/2d-decompositions-overview.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+In PyWavelets it is implemented for n-dimensional data by `wavedecn` and the
+inverse by `waverecn`.  1D and 2D versions of these routines also exist.
+
+.. image:: ../_static/Mallat.png
```
Comment (rgommers) [inline doc/source/ref/2d-decompositions-overview.rst]: again would be good to commit the scripts rather than the pngs. do you still have them?

**Comment by rgommers** [comment]
except for the couple of TODOs, this looks ready to me

File: doc/source/ref/2d-decompositions-overview.rst (hunk 3 above)
Comment (grlee77) [reply doc/source/ref/2d-decompositions-overview.rst]: I made these in inkscape, but at some later point I did make a Matlab function that could create these type of plots.  I will look into using that instead

Comment (grlee77) [reply doc/source/ref/2d-decompositions-overview.rst]: I made these in inkscape, but at some later point I did make a Matlab function that could create these type of plots.  I will look into using that instead

**Comment by grlee77** [comment]
This now generates the figures nicely on the fly, but I had to duplicate some utility code in the two scripts.  One alternative would be to store these plotting utilities elsewhere, perhaps within a private module such as ``pywt._doc_utils.py`` and import from there (I don't think we want to maintain them as part of the public API).  That would make these two scripts much shorter. What do you think?

It is also to make multiple plots with a single script but then I think they would have to appear adjacent in the sphinx document rather then separated as they are now.

**Comment by rgommers** [comment]
`pywt._doc_utils.py ` sounds like a good idea to me. Do you want to do that in this PR, or leave it for later?

**Comment by grlee77** [comment]
I will try separating out to `_doc_utils.py` in a separate PR

**Pull request merged**